### ISSUE TITLE
NugetPack .csproj + project.json: simple mappings not working

### ISSUE THREAD

**Issue by BladeWise** [NuGet/Home#3184]
As was discussed in [this comment](https://github.com/NuGet/Home/issues/3145#issuecomment-233994663), defining a project.json with this simple mapping

``` json
"packOptions": {
    "files": {
      "mappings": {
        "mylib": {
          "include": [
            "bin/Debug/*.dll"
          ]
        }
      }
    }
  }
```

does not work, since it does not create the mylib folder, and no dll file is included.
To achieve the expected behavior a small trick can be used:

``` json
"packOptions": {
    "files": {
      "mappings": {
        "mylib": {
          "include": [
            "bin/Debug*/*.dll"
          ]
        }
      }
    }
  }
```

**Comment by MistyKuu** [comment]
@BladeWise one of the stars has been eaten by editor, please edit your post

**Comment by BladeWise** [comment]
@MistyKuu Done.

**Comment by rrelyea** [comment]
NuGet pack will probably not work with project.json mappings ever. As we'll be moving to csproj away from xproj.

**Comment by BladeWise** [comment]
It does work, at least partially.
The fact is, if NetStandard projects can only use project.json, and support is only partial, how are we supposed to produce NetStandard packages before such functionalites are migrated to csproj? What is the 'proper' walkthrough? Avoid .csproj + project.json?

Not really happy about a won't fix...

**Comment by MistyKuu** [comment]
@BladeWise, @rrelyea  agreed 100%. This is a basic functionality. I was recommended to use csproj + project.json as migration to new csproj should be easier. Now I am struggling with current configuration. Adding files into nuget packages is implemented the only logic that needs to be fixed is to properly map it into directory structure so I suspect it doesn't involve much work on your side.

**Comment by MistyKuu** [comment]
@rrelyea I don't understand a decision to drop support for nuspec in csproj + project.json files in 3.5 RTM. The problem with dependencies has been fixed but because of that we get a huge cut in functionalities. Nuspec provided a lot more possibilities than project.json

**Comment by rohit21agrawal** [comment]
@MistyKuu you can still author a nuspec file in your project directory and pack using nuget pack *.nuspec ...maybe that will unblock some of your scenarios?

**Comment by rohit21agrawal** [comment]
@BladeWise @MistyKuu just found out that you need to use this json to get your scenarios to work as expected on Windows :

```
"packOptions": {
    "files": {
      "mappings": {
        "mylib": {
          "include": [
            "bin\\Debug\\*.dll"
          ]
        }
      }
    }
  }
```

Note the difference in the directory separator character. 

Would you be able to try the same technique on #3173 and let us know if that solves it as well?

**Comment by MistyKuu** [comment]
@rohit21agrawal it works - good work! But it looks like it doesn't work for #3173

**Comment by rohit21agrawal** [comment]
#3173 wirks with includeFiles from what i remember of my testing

**Comment by BladeWise** [comment]
@rohit21agrawal Yes, it seems that current RC works using double backslashes, provided that includeFiles is used. It seems that includeFiles replaces *\* usage (I still preferred the old syntax, tho :)).

The only notable issue I have found with this version is that using this kind of mapping

``` json
    "files": {
      "mappings": {
        "lib/net40": {
          "include": [
            "bin\\Debug\\net40\\Sample.dll"
          ],
          "includeFiles": [
            "bin\\Debug\\net40\\**\\Sample.resources.dll"
          ]
        }
      }
    }
```

`nuget pack` complains about

```
WARNING: Issue found with package 'Sample'.

WARNING: Issue: Assembly outside lib folder.
WARNING: Description: The assembly 'lib/net40\Sample.dll' is not inside the 'lib' folder and hence it won't be added as reference when the package is installed into a project.
WARNING: Solution: Move it into the 'lib' folder if it should be referenced.
```
